File: app/src/main/java/org/digitalcampus/oppia/model/CustomValue.java
package org.digitalcampus.oppia.model;

import androidx.annotation.NonNull;

public class CustomValue<T> {

    private T value;

    public CustomValue(T value) {
        setValue(value);
    }

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        if (value instanceof String || value instanceof Boolean || value instanceof Integer || value instanceof Float) {
            this.value = value;
        } else {
            throw new IllegalArgumentException("value must one of these types: String, int, float or boolean");
        }
    }

    @NonNull
    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: app/src/main/java/org/digitalcampus/oppia/model/CustomValue.kt
package org.digitalcampus.oppia.model

sealed class CustomValueType
data class StringValue(val value: String) : CustomValueType()
data class BooleanValue(val value: Boolean) : CustomValueType()
data class IntValue(val value: Int) : CustomValueType()
data class FloatValue(val value: Float) : CustomValueType()

class CustomValue(private val value: CustomValueType) {
    fun getValue(): Any {
        return value
    }

    override fun toString(): String {
        return value.toString()
    }
}